packages/kbn-i18n/src/core/i18n.ts

```diff
@@ -77,13 +78,17 @@ export function addMessages(newMessages: PlainMessages = {}, locale = newMessage
   messages[normalizedLocale] = {
```

Comment:
__issue:__ it's getting harder to read this code, let's change to something like this:
```js
const existingTranslation = translationsForLocale[normalizedLocale] || { messages: {} };
  translationsForLocale[normalizedLocale] = {
    // Do we need to merge formats themselves? Also it's a bit weird, we don't update global `formats`..?
    formats: newTranslation.formats || existingTranslation.formats,
    locale: newTranslation.locale || existingTranslation.locale,
    messages: {
      ...existingTranslation.messages,
      ...(newTranslation.messages || {}),
    },
  };
```

Also is there any reason why we can't make `messages` required property on `Translation` interface, it feels logical to me, as there any issues with that? It will allow us to remove a bunch of `if` here and there.